client/src/partials/journal/journal.js

```diff
@@ -58,37 +66,115 @@ function JournalController(Transactions, Sorting, Grouping, Pagination, Filterin
+      { field : 'currency_id', displayName : 'Currency', visible: false },
+      
+      // @todo this should be formatted showing the debitor/credior
+      { field : 'deb_cred_uuid', displayName : 'Recipient', visible : false }, 
+      { field : 'deb_cred_type', displayName : 'Recipient Type', visible : false }, 
+
+      // @fixme inv_po_id -> reference
+      { field : 'inv_po_id', displayName : 'Reference Document', visible : false },
+      { field : 'user', displayName : 'Responsible', visible : false },
+      
+      // @fixme this field should not come from the database as 'cc'
+      { field : 'cc', displayName : 'Cost Center', visible : false },
+      { field : 'pc', displayName : 'Profit Center', visible : false }
+    ];
+
+    vm.gridOptions.columnDefs = cache.columns ? resetColumns(cache.columns) : columns;
+
+    // This function opens a modal to let the user show or Hide columns
+    vm.openColumnConfigModal = function openColumnConfigModal() {
+
+      var instance = Modal.open({
+        templateUrl: 'partials/journal/modals/columnsConfig.modal.html',
+        controller:  'ColumnsConfigModalController as ColumnsConfigModalCtrl',
+        size:        'md',
+        backdrop:    'static',
+        animation:   true,
+        resolve:     {
+          columnList:  function columnListProvider() { return vm.gridOptions.columnDefs; },
+          defaultColumns : function defaultColumnListProvider () { return angular.copy(columns);}
+        }
+      });
+
+      instance.result.then(function (result) {
+        vm.gridOptions.columnDefs = result.columns;
+        columnConfig.refreshColumns();
+        cache.columns = result.columns;
+      });
+    };
+
+  /**
+  * This method is there to fix problem caused by fetching column from cache
+  * When columns are fetched from cache, sorting functionnality does not work
+  * this function is there to reset this functionnality
+  *
+  * @todo This is a lazy solution, we must refactor our cache module or redesign our posting journal module
+  **/  
+  function resetSorting (){
+    cache.columns.map(function (item){
+      if(item.sort){
+        item.sortingAlgorithm = sorting.transactionIds;
+        return item;
+      }
+      return item;
+    });
+  }
+
+  /**
+  * This method is there to fix problem caused by fetching column from cache
+  * When columns are fetched from cache, filtering functionnality does not work
+  * this function is there to reset this functionnality
+  *
+  * @todo This is a lazy solution, we must refactor our cache module or redesign our posting journal module
+  **/ 
+  function resetFiltering (){
+    cache.columns.map(function (item){
+      if(item.sort){
+        item.sortingAlgorithm = sorting.transactionIds;
+        return item;
+      }
+      return item;
+    });
+  }
+
+  /** 
+  * A facade to call resetSorting and resetFiltering method
+  * @return {object} array of columns
+  **/
+
+  function resetColumns (){
```

Comment:
Can the `reset...()` functions be incorporated into their own services?  For example, instead of `resetFiltering()` or `resetSorting()`, could we just write `sorting.reset()` or `filtering.reset()` instead?

That way, each service would be responsible for their own resets, and the controller wouldn't have as much code.